src/jit/target.h

```diff
@@ -1884,11 +1892,58 @@ inline bool         genIsValidDoubleReg(regNumber reg)
 #endif // defined(LEGACY_BACKEND) && defined(_TARGET_ARM_)
 
 /*****************************************************************************
+*
+*  Returns true if our target architecture uses a fixed return buffer register
+*/
+inline bool         hasFixedRetBuffReg()
+{
+    // Disable this until the VM changes are also enabled
+#if 0 //def _TARGET_ARM64_
+    return true;
+#else
+    return false;
+#endif
+}
+
+/*****************************************************************************
+*
+*  Returns the regNumber for the fixed return buffer 
+*/
+inline regNumber    theFixedRetBuffReg()
+{
+    assert(hasFixedRetBuffReg());   // This predicate should be checked before calling this method
+#ifdef _TARGET_ARM64_
+    return REG_ARG_RET_BUFF;
+#else
+    return REG_NA;
+#endif
+}
+
+/*****************************************************************************
+*
+*  Returns the argNum for the fixed return buffer 
+*/
+inline unsigned     theFixedRetBuffArgNum()
+{
+    assert(hasFixedRetBuffReg());   // This predicate should be checked before calling this method
+#ifdef _TARGET_ARM64_
+    return RET_BUFF_ARGNUM;
+#else
+    return (unsigned) -1;
+#endif
+}
+
+/*****************************************************************************
  *
  *  Returns true if the register is a valid integer argument register 
+ *  This method also returns true on Arm64 when 'reg' is the RetBuff register 
  */
 inline bool         isValidIntArgReg(regNumber reg)
 {
+    if (hasFixedRetBuffReg() && (reg == theFixedRetBuffReg()))
+    {
+        return true;
+    }
     return (genRegMask(reg) & RBM_ARG_REGS) != 0;
```

Comment:
I think the case of genRegMask(reg) & RBM_ARG_REGS would be the common case considering methods that return retbuf would be fewer in number overall.  It would be a good idea to check for second condition first.

e.g.
if((genRegMask(reg) & RBM_ARG_REGS) != 0)
{
    return true
}

return hasFixedRegBuffReg() && (reg == theFixedRetBufReg());